portability: cmsis: Reduce control block size by not copying object's name

utsavm9 opened the pull request:
## Description

All CMSIS-RTOSv2 objects can have a human-readable name to aid in debugging. However, copying the name string instead of storing a pointer to it bloats up the size of control block, and is not needed.

Changes:
* Defer thread name implementation onto Zephyr Thread API [`k_thread_name_set/get()`](https://docs.zephyrproject.org/apidoc/latest/group__thread__apis.html#gadebf45da56dee393164569742459dc0a).
* Allow all `os***GetName()` to work from ISRs, as expected by [CMSIS-RTOSv2 spec](https://arm-software.github.io/CMSIS_6/main/RTOS2/group__CMSIS__RTOS__ThreadMgmt.html#gac3230f3a55a297514b013ebf38f27e0a).
*  For all other objects, store thread name as a pointer passed by initialization attributes.

After this PR, the test application `tests/subsys/portability/cmsis_rtos_v2` takes up 144 less bytes of flash and 956 less bytes of RAM.

## Test Plan

* All tests pass locally when running with `west build tests/subsys/portability/cmsis_rtos_v2 --board=nrf52dk/nrf52832 --pristine && west flash`

utsavm9 committed:
portability: cmsis: Store thread name within Zephyr k_thread

Use underlying Zephyr thread directly to store thread name instead of
storing the name in CMSIS control block. Also, allow `osThreadGetName`
to work within ISR, as expected from spec.

Signed-off-by: Utsav Munendra <utsavm@meta.com>

utsavm9 commented:
Requesting review from @peter-mitsis and @cfriedt too as they have been a great help for reviewing my similar PRs. Consider adding them @kartben.

cfriedt reviewed (dismissed):
Preemptive +1 - looks ok to me. Hopefully CI agrees!

utsavm9 commented:
Looking into failing CI tests...

utsavm9 committed:
portability: cmsis: Avoid copying objects names into control block

Instead, just store the pointer to the string provided as part of the
RTOS object init attributes.

Signed-off-by: Utsav Munendra <utsavm@meta.com>

utsavm9 commented:
It took me a while to find out that leaving the name pointer to `NULL` causes some of the twister tests to fail unexpectedly. This does not happen on my board, so I am still not sure why.

Therefore, always setting the name to a valid string if no name was provided by the user. Tests should pass now, PR is ready, requesting review again.

utsavm9 commented:
Thanks for the quick review :)
